src/Microsoft.VisualStudio.Extensibility.Testing.Xunit.Shared/Harness/VisualStudioInstanceKey.cs

```diff
@@ -53,22 +60,54 @@ public override int GetHashCode()
             hashCode = (hashCode * -1521134295) + Version.GetHashCode();
             hashCode = (hashCode * -1521134295) + EqualityComparer<string>.Default.GetHashCode(RootSuffix);
             hashCode = (hashCode * -1521134295) + MaxAttempts.GetHashCode();
+            hashCode = (hashCode * -1521134295) + EnvironmentVariables.Count.GetHashCode();
             return hashCode;
         }
 
         public string SerializeToString()
         {
-            return $"{(int)Version};{RootSuffix};{MaxAttempts}";
+            var builder = new StringBuilder();
+            builder.Append(EnvironmentVariables.Count);
+            foreach (var environmentVariable in EnvironmentVariables)
+            {
+                builder.Append(';');
+                builder.Append(Convert.ToBase64String(Encoding.UTF8.GetBytes(environmentVariable)));
```

Comment:
Just to confirm - its not necessary for this to be human readable?

Reply:
From the code I feel it might because there can't be any `;` in the serialized string. Otherwise it might affect the Parsing below.
Another way might be to move to Json. But in that case, we need to load Json dlls.

Reply:
This doesn't need to be human readable (it's for XUnit to communicate with the test host). I just didn't want user strings here to cause a problem (and yes I now see that Root Suffix is added without encoding, but that seems less likely to be problematic).